### PR TITLE
Adapt shorthand syntax for all option-less dependencies

### PR DESCRIPTION
**Pull request by TheLartians**

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -23,13 +23,7 @@ else()
...
 endif()
 
-CPMAddPackage(
-  NAME Format.cmake
-  GITHUB_REPOSITORY TheLartians/Format.cmake
-  VERSION 1.6
-  OPTIONS # enable cmake formatting
-          "FORMAT_CHECK_CMAKE ON"
-)
+CPMAddPackage("gh:TheLartians/Format.cmake@1.7.0")
```
Comment (ClausKlein) [inline test/CMakeLists.txt:26]: what is about the `OPTIONS`?

**Comment by ClausKlein** [comment]
@TheLartians this breaks the tests!
```bash
+ cmake -S all -B build/all -G Ninja -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_BUILD_TYPE=Debug -DCMAKE_PREFIX_PATH=/Users/clausklein/Workspace/cpp/ModernCppStarter/stage -DTEST_INSTALLED_VERSION=1 -DENABLE_TEST_COVERAGE=1
-- The CXX compiler identification is Clang 11.1.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/local/opt/llvm/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- cxxopts version 2.2.0
-- The C compiler identification is Clang 11.1.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/local/opt/llvm/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- CPM: adding package cxxopts@2.2.0 (v2.2.0 at /Users/clausklein/.cache/CPM/cxxopts/d5c328043c71a6a5670c290d8646b093756db962)
-- CPM: adding package Greeter@ (/Users/clausklein/Workspace/cpp/ModernCppStarter/standalone/..)
-- CPM: Greeter: adding package PackageProject.cmake@1.4.1 (v1.4.1 at /Users/clausklein/.cache/CPM/packageproject.cmake/ca5861b6aff5463f0c3bee585adf35e3d2defff7)
-- Version: 7.1.3
-- Build type: Debug
-- CXX_STANDARD: 11
-- Performing Test has_std_11_flag
-- Performing Test has_std_11_flag - Success
-- Performing Test has_std_0x_flag
-- Performing Test has_std_0x_flag - Success
-- Performing Test SUPPORTS_USER_DEFINED_LITERALS
-- Performing Test SUPPORTS_USER_DEFINED_LITERALS - Success
-- Performing Test FMT_HAS_VARIANT
-- Performing Test FMT_HAS_VARIANT - Success
-- Required features: cxx_variadic_templates
-- Performing Test HAS_NULLPTR_WARNING
-- Performing Test HAS_NULLPTR_WARNING - Success
-- Looking for strtod_l
-- Looking for strtod_l - found
-- CPM: Greeter: adding package fmt@7.1.3 (7.1.3 at /Users/clausklein/.cache/CPM/fmt/9dd47b889fbf7c876d24be56ca097a884004b789)
-- CPM: adding package doctest@2.4.5 (2.4.5 at /Users/clausklein/.cache/CPM/doctest/5a9c0f357efafb61890e91b445aa0f5d4c2bca1b)
-- Found Python: /usr/local/Frameworks/Python.framework/Versions/3.9/bin/python3.9 (found version "3.9.1") found components: Interpreter 
-- CPM: adding package Format.cmake@1.7.0 (v1.7.0 at /Users/clausklein/.cache/CPM/format.cmake/e25a26778e5810b3fa49523bf8cadf5f71bacc26)
-- CPM: adding package m.css@0 (42d4a9a48f31f5df6e246c948403b54b50574a2a to /Users/clausklein/.cache/CPM/m.css/ce3daea984872362c3a8ed1c3d8956adbc400a88)
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/clausklein/Workspace/cpp/ModernCppStarter/build/all
+ cmake --build build/all
[0/2] Re-checking globbed directories...
[10/10] Linking CXX executable standalone/Greeter
+ cmake --build build/all --target test
[0/2] Re-checking globbed directories...
[0/1] Running tests...
Test project /Users/clausklein/Workspace/cpp/ModernCppStarter/build/all
    Start 1: options
Could not find executable options_test
Looked in the following places:
options_test
options_test
Release/options_test
Release/options_test
Debug/options_test
Debug/options_test
MinSizeRel/options_test
MinSizeRel/options_test
RelWithDebInfo/options_test
RelWithDebInfo/options_test
Deployment/options_test
Deployment/options_test
Development/options_test
Development/options_test
Unable to find executable: options_test
1/5 Test #1: options ..........................***Not Run   0.00 sec
    Start 2: find-package-test
2/5 Test #2: find-package-test ................   Passed    4.23 sec
    Start 3: add-subdirectory-test
3/5 Test #3: add-subdirectory-test ............   Passed    3.47 sec
    Start 4: Greeter
4/5 Test #4: Greeter ..........................   Passed    0.01 sec
    Start 5: Greeter version
5/5 Test #5: Greeter version ..................   Passed    0.01 sec

80% tests passed, 1 tests failed out of 5

Total Test time (real) =   7.71 sec

The following tests FAILED:
	  1 - options (Not Run)
Errors while running CTest
FAILED: CMakeFiles/test.util 
cd /Users/clausklein/Workspace/cpp/ModernCppStarter/build/all && /usr/local/Cellar/cmake/3.19.5/bin/ctest --force-new-ctest-process
ninja: build stopped: subcommand failed.
bash-3.2$ 
```

File: test/CMakeLists.txt (hunk 1 above)
Comment (TheLartians) [reply test/CMakeLists.txt:26]: `FORMAT_CHECK_CMAKE` is enabled by default in Format.cmake after version 1.7.0.

**Comment by TheLartians** [comment]
Huh, it seems the tests leaked from `cxxopts`, even though they were added using `EXCLUDE_FROM_ALL`. Imo this is a CMake bug, but we should probably explicitly disable the option again in the starter. Until then tests in the `all` should be run by calling the `./build/all/test/GreeterTests` executable directly.

**Comment by ClausKlein** [comment]
there is more strange: 
```bash
bash-3.2$ grep -w file build/standalone/compile_commands.json 
  "file": "/Users/clausklein/Workspace/cpp/ModernCppStarter/standalone/source/main.cpp"
  "file": "/Users/clausklein/.cache/CPM/cxxopts/d5c328043c71a6a5670c290d8646b093756db962/src/example.cpp"
  "file": "/Users/clausklein/Workspace/cpp/ModernCppStarter/source/greeter.cpp"
  "file": "/Users/clausklein/.cache/CPM/fmt/9dd47b889fbf7c876d24be56ca097a884004b789/src/format.cc"
  "file": "/Users/clausklein/.cache/CPM/fmt/9dd47b889fbf7c876d24be56ca097a884004b789/src/os.cc"
bash-3.2$ 
```
Why are this files from CPM cache include in the `compile-comands.json` file?

I do not want to check them with `run-clang-tidy`!

This was the main reason why I prefer `CMP.cmake` against `native fetch contents`, which clone the sources into `build tree`.

**Comment by ClausKlein** [comment]
same for all package:
```
bash-3.2$ grep -w file build/all/compile_commands.json 
  "file": "/Users/clausklein/Workspace/cpp/ModernCppStarter/standalone/source/main.cpp"
  "file": "/Users/clausklein/.cache/CPM/cxxopts/d5c328043c71a6a5670c290d8646b093756db962/src/example.cpp"
  "file": "/Users/clausklein/Workspace/cpp/ModernCppStarter/source/greeter.cpp"
  "file": "/Users/clausklein/.cache/CPM/fmt/9dd47b889fbf7c876d24be56ca097a884004b789/src/format.cc"
  "file": "/Users/clausklein/.cache/CPM/fmt/9dd47b889fbf7c876d24be56ca097a884004b789/src/os.cc"
  "file": "/Users/clausklein/Workspace/cpp/ModernCppStarter/test/source/greeter.cpp"
  "file": "/Users/clausklein/Workspace/cpp/ModernCppStarter/test/source/main.cpp"
  "file": "/Users/clausklein/.cache/CPM/doctest/5a9c0f357efafb61890e91b445aa0f5d4c2bca1b/doctest/parts/doctest.cpp"
bash-3.2$ 
```

**Pull request merged**